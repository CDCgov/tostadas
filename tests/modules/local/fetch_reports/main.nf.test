nextflow_process {

    name "Test Process FETCH_REPORTS"
    script "modules/local/fetch_reports/main.nf"
    process "FETCH_REPORTS"

    test("Run with test profile and mock batch") {

        when {
            params {
                submission_prod_or_test = "test"
                dry_run = true
                submission_mode = "ftp"
                metadata_basename = "rsv_test_metadata" 
                species = "rsv"
            }

            process {
                """
                // Inputs
                input[0] = [ [ batch_id: 'batch_1' ], file("${projectDir}/tests/modules/local/fetch_reports/batch_1/") ]
                input[1] = file("${projectDir}/conf/submission_config.yaml")
                """
            }
        }

        then {
            assert process.success

            
        }
    }
}
