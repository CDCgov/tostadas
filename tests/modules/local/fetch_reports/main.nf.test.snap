{
    "Run with test profile and mock batch": {
        "content": [
            "fetch_submission.log:md5,fc75830516ca9996550bbdb6dc99ebdb"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:21:40.356775611"
    }
}